src/distributions/uniform.rs

```diff
@@ -494,6 +496,86 @@ macro_rules! uniform_float_impl {
 uniform_float_impl! { f32, 32 - 23, next_u32 }
 uniform_float_impl! { f64, 64 - 52, next_u64 }
 
+/// Implementation of `UniformImpl` for `Duration`.
+/// 
+/// Unless you are implementing `UniformImpl` for your own types, this type should
+/// not be used directly, use `Uniform` instead.
+#[cfg(feature = "std")]
+#[derive(Clone, Copy, Debug)]
+pub struct UniformDuration {
+    offset: Duration,
+    mode: UniformDurationMode,
+}
+
+#[derive(Debug, Copy, Clone)]
+enum UniformDurationMode {
+    Small {
+        nanos: Uniform<u64>,
+    },
+    Large {
+        size: Duration,
+        secs: Uniform<u64>,
+        nanos: Uniform<u32>,
```

Comment:
Maybe I should have been a bit more clear, and the documentation is being worked on...

If the value is constant, LLVM should be able to figure it out and do things at compile time. So you would only need to cache `secs`, `nanos` can just use `gen_range`.

Reply:
Ah right, fixed!

Reply:
Correct me if I'm wrong but `gen_range` will produce a smaller `zone` for rejection sampling than `Uniform::new`.

Shouldn't this then use `Uniform::sample` rather than `sample_single` as long as the range calculations are all done at compile time?

(Relevant documentation changes here https://github.com/rust-lang-nursery/rand/pull/480)

Reply:
Good point, that can be a small improvement. Do you want to make a PR?